tests/dogtag/pytest-ansible/pytest/performance_test/test_cert_revocation.py

```diff
@@ -0,0 +1,79 @@
+"""
+# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+#
+#   Description: Load Test client Server Side configuration
+#   Steps :-
+#
+# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+#
+#   Author: Deepak Punia <dpunia@redhat.com>
+#
+# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+# Copyright Red Hat, Inc.
+# SPDX-License-Identifier: GPL-2.0-or-later
+# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+"""
+import re
+import sys
+import logging
+import threading
+import argparse
+from pki.client import PKIConnection
+from pki.cert import CertClient
+
+parser = argparse.ArgumentParser()
+parser.add_argument("--hostname", help="<CA hostname>")
+parser.add_argument("--port", help="<CA Port Number>")
+parser.add_argument("--client-cert", help="path for admin.pem certificate")
+parser.add_argument("--number-of-clients", help="Number of thread", type=int)
+parser.add_argument("--cert_sn_file", help="path for certificate serial number file")
+
+log = logging.getLogger()
+logging.basicConfig(stream=sys.stdout, level=logging.INFO)
+
+args = parser.parse_args()
+
+# Create a PKIConnection object that stores the details of the CA.
+connection = PKIConnection('https', args.hostname, args.port)
+
+# The pem file used for authentication. Created from a p12 file using the
+# command -
+# openssl pkcs12 -in <p12_file_path> -out /tmp/auth.pem -nodes
+connection.set_authentication_cert(args.client_cert)
+
+# Instantiate the CertClient
+cert_client = CertClient(connection)
+
+
+def run_test(cert_sn_file):
+    # execute the specified number of tests
+    cert_sn = cert_sn_file.readlines()
+    for cert_sn in cert_sn:
+        try:
+            cert_sn = re.findall('Serial Number:.*', cert_sn)[0].split(":")[1].strip()
+            revoke_data = cert_client.revoke_cert(cert_sn, revocation_reason='Key_Compromise')
+            if revoke_data.operation_result == 'success':
```

Comment:
It should raise an exception if `revoke_data.operation_result != 'success'`.

Reply:
Added